### PR TITLE
Fix star imports f2py

### PR DESCRIPTION
**Pull request by charris**
Also bite the bullet and make PEP8 fixes in `numpy/f2py/*.py`. That cleans up the code remarkably, leaving
it less incomprehensible. Some errors and potential errors were also detected in fixing the `import *` imports.

### CODE REVIEW

File: numpy/f2py/auxfuncs.py [hunk 1]
```diff
@@ -40,44 +65,57 @@ def outmess(t):
...
         sys.stdout.write(t)
 
+
 def debugcapi(var):
     return 'capi' in debugoptions
 
+
 def _isstring(var):
-    return 'typespec' in var and var['typespec']=='character' and (not isexternal(var))
+    return 'typespec' in var and var['typespec'] == 'character' and (not isexternal(var))
```
Comment (jaimefrio) [inline numpy/f2py/auxfuncs.py]: This fits in 79 columns?

File: numpy/f2py/auxfuncs.py [hunk 2]
```diff
@@ -88,187 +126,221 @@ def get_kind(var):
...
+    return not ('dimension' not in var or isstring(var))
+
 
 def ismoduleroutine(rout):
     return 'modulename' in rout
 
+
 def ismodule(rout):
-    return ('block' in rout and 'module'==rout['block'])
+    return ('block' in rout and 'module' == rout['block'])
```
Comment (jaimefrio) [inline numpy/f2py/auxfuncs.py]: No parenthesis needed around the return.

File: numpy/f2py/auxfuncs.py [hunk 3]
```diff
@@ -88,187 +126,221 @@ def get_kind(var):
...
+
 
 def isfunction(rout):
-    return ('block' in rout and 'function'==rout['block'])
+    return ('block' in rout and 'function' == rout['block'])
```
Comment (jaimefrio) [inline numpy/f2py/auxfuncs.py]: Extra () here again

File: numpy/f2py/auxfuncs.py [hunk 4]
```diff
@@ -88,187 +126,221 @@ def get_kind(var):
...
+
+# def isfunction_wrap(rout):
```
Comment (jaimefrio) [inline numpy/f2py/auxfuncs.py]: Isn't the policy to remove commented out code?

File: numpy/f2py/auxfuncs.py [hunk 5]
```diff
@@ -88,187 +126,221 @@ def get_kind(var):
...
 
 def isfunction_wrap(rout):
     if isintent_c(rout):
         return 0
     return wrapfuncs and isfunction(rout) and (not isexternal(rout))
 
+
 def issubroutine(rout):
-    return ('block' in rout and 'subroutine'==rout['block'])
+    return ('block' in rout and 'subroutine' == rout['block'])
```
Comment (jaimefrio) [inline numpy/f2py/auxfuncs.py]: Extra ()

**Comment by rgommers** [comment]
New, now only 88.9% incomprehensible:)

More seriously, thanks for taking on all this housekeeping Chuck.

File: numpy/f2py/auxfuncs.py (hunk 1 above)
Comment (charris) [inline numpy/f2py/auxfuncs.py]: No:) I ignored long lines for this PR, some are well over a 100 characters.

File: numpy/f2py/auxfuncs.py (hunk 4 above)
Comment (charris) [inline numpy/f2py/auxfuncs.py]: Yes, but with this code I'm never sure if commented out code might be informative as to intent.

**Comment by charris** [comment]
Still many long lines, mostly strings. Breaking those is a lot of somewhat delicate work and more than I want to do at this time.

**Pull request merged**